main/src/util.ts

```diff
@@ -31,3 +32,18 @@ export function getAppVersion(): string {
 
   return getVersionFromGit()
 }
+
+export async function pollWindowReady(window: BrowserWindow): Promise<void> {
+  return new Promise((resolve) => {
+    const checkReady = () => {
+      if (window?.isVisible() && window.webContents.isLoading() === false) {
+        log.info('Window is ready and visible')
+        resolve(void 0)
+      } else {
+        log.info('Window not ready yet, waiting...')
+        setTimeout(checkReady, 100)
```

Comment:
I guess the async sleep function I suggested could already be reused in this function :thinking:  probably we could entirely remove the `new Promise`thing from here